src/coredump/coredump.c

```diff
@@ -186,7 +186,7 @@ static int fix_acl(int fd, uid_t uid) {
                 return 0;
 
         /* Make sure normal users can read (but not write or delete) their own coredumps */
-        r = add_acls_for_user(fd, uid);
+        r = fd_add_uid_acl_permission(fd, uid, /* read = */ true, /* write = */ false, /* execute = */ false);
```

Comment:
S_IREAD is V7 UNIX btw, POSIX calls them S_IRUSR.

But I think these macros suck for this purpose, since according to the docs they are the access mode for the file *owner*, but this doesn't really apply here, we set the access mode for arbitrary users here after all. S_IRUSR + S_IWUSR + S_IXUSR are defined as 0400, 0200, 0100, i.e. encode the file "owner" thing, they don't least-significant bits, but some in the middle, which I found confusing and misleading in this context here.

Reply:
hmm, ACL_READ/ACL_WRITE/ACL_EXECUTE might be an option, but it's ugly too: in the Linux implementation these are a bitmask, i.e. defined to 4, 2, 1. The `acl_add_perm()` API call accepts them as bitmask too. so sounds great. except that the docs say pretty explicitly you may only pass one value at a time, i.e. it's not officially a bitmask. And the (withdrawn) POSIX spec for this does not suggest it was a mask either (see https://manualzz.com/doc/13559599/posix_1003.1e), but explicitly says it can be mapped to anything.

So yeah, hmpf. They don't sound so great either. in practice they'd work, since they are a bitmask, but according to every api documentaiton i could find they are not a bitmask. 

Which makes me a bit nervous using them. I mean I am all for accepting reasonable Linux specific implementations as gold standard. But going directly against documentation like this is a bit much. dunno?

That's the reason why i just used three separate bool function arguments, to sidestep the whole issue.

Reply:
(in the freebsd implemenation they are a bitmask too and `acl_add_perm()`'s code accepts them as such, but also documented as not being a bitmask. So i guess it would even be portable to assume they are a bitmask, if we cared about that, except that docs say otherwise...)

Reply:
https://github.com/systemd/systemd/pull/16841